src/main/java/org/zone/region/flag/entity/player/move/preventing/PreventPlayersListener.java

```diff
@@ -53,4 +58,142 @@ public void onPlayerMove(MoveEntityEvent event, @Getter("entity") Player player)
         event.setCancelled(true);
     }
 
+    @Listener
+    public void onPlayerLoginEvent(ServerSideConnectionEvent.Join joinevent) {
+        ServerPlayer serverPlayer = joinevent.player();
+        Optional<Zone> opZone = ZonePlugin
+                .getZonesPlugin()
+                .getZoneManager()
+                .getPriorityZone(joinevent.player().world(), joinevent.player().position());
+
+        if (opZone.isEmpty()) {
+            return;
+        }
+
+        if (!(opZone.get().containsFlag(FlagTypes.PREVENT_PLAYERS))) {
+            return;
+        }
+
+        Optional<PreventPlayersFlag> flag = opZone
+                .get()
+                .getFlag(FlagTypes.PREVENT_PLAYERS);
+        if (flag.isEmpty()) {
+            return;
+        }
+        if (flag.get().hasPermission(opZone.get(),serverPlayer.uniqueId())) {
+            Optional<Vector3i> opTeleportPos = opZone
```

Comment:
You know I hate nested if statements. 

```java
        if (!flag.get().hasPermission(opZone.get(),serverPlayer.uniqueId())) {
            return;
        }
```

reads so much more clear

Reply:
XD

Reply:
emm so else not needed?